sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java

```diff
@@ -164,14 +167,34 @@ public <T> SpanBuilder setAttribute(AttributeKey<T> key, T value) {
   }
 
   @Override
-  public SpanBuilder setStartTimestamp(long startTimestamp, TimeUnit unit) {
+  @SuppressWarnings("unchecked")
+  public ExtendedSpanBuilder setAllAttributes(Attributes attributes) {
+    if (attributes == null || attributes.isEmpty()) {
+      return this;
+    }
+    attributes.forEach(
+        (attributeKey, value) -> setAttribute((AttributeKey<Object>) attributeKey, value));
+    return this;
+  }
```

Comment:
If we add this here then we should also remove the default method implementation from `SpanBuilder`. Alternatively we could leave it as default method in `SpanBuilder` and add
```java
  @Override
  default ExtendedSpanBuilder setAllAttributes(Attributes attributes) {
    return (ExtendedSpanBuilder) SpanBuilder.super.setAllAttributes(attributes);
  }
```
to `ExtendedSpanBuilder`